modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx

```diff
@@ -314,12 +314,20 @@ export default function ObjectFieldFormBase({
 					error={errors.output}
 					label={Liferay.Language.get('output')}
 					onChange={({label, value}) => {
-						setValues({
-							objectFieldSettings: [
+						let newObjectFieldSettings: ObjectFieldSetting[] = [];
+
+						if (values.objectFieldSettings) {
+							newObjectFieldSettings = [
 								...(values.objectFieldSettings?.filter(
 									(objectFieldSetting) =>
 										objectFieldSetting.name !== 'output'
 								) as ObjectFieldSetting[]),
+							];
+						}
```

Comment:
Hey @seltong, in this case as you are doing a simple filter and only the filter result will be the `newObjectFieldSettings`, we don't need to use the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), just pass the filter return directly:

```tsx
newObjectFieldSettings = values.objectFieldSettings?.filter(
    (objectFieldSetting) =>
	    objectFieldSetting.name !== 'output'
) as ObjectFieldSetting[];
```